tests/python/unittest/test_tir_schedule_transform_layout.py

```diff
@@ -329,5 +329,302 @@ def test_transform_block_layout_fail_mixed_iter_type(use_block_name):
         )
 
 
+class BasePaddingCompare(tvm.testing.CompareBeforeAfter):
+    pad_value = tvm.testing.parameter(None)
+
+    transformed_buffer = tvm.testing.parameter("A")
+
+    @pytest.fixture
+    def transform(self, pad_value, transformed_buffer):
+        def transform(mod):
+            sch = tir.Schedule(mod)
+            sch.transform_layout(
+                "block", transformed_buffer, lambda i: [i // 4, i % 4], pad_value=pad_value
+            )
+            # sch.transform_block_layout("block", lambda i: [i // 4, i % 4])
+            return sch.mod
+
+        return transform
+
+
+class TestNoPadding(BasePaddingCompare):
+    """Transformations without padding do not depend on pad_value."""
+
+    pad_value = tvm.testing.parameter(None, 42)
+
+    def before():
```

Comment:
It's great if we can support padding opaque blocks. However, non-opaque blocks are the most common cases. 

Could you please change the test cases into non-opaque blocks?
e.g.
```python
with T.block():
    vi = T.axis.remap("S", [i])
    A[vi] = 0
```

Reply:
For this specific instance, the test case for non-opaque blocks is lower.  I've re-ordered these test cases so that the non-opaque block tests occur first, and are opaque blocks are the ones called out as being non-standard.

I am in the process of updating all remaining examples in the tests to use non-opaque blocks, though I expect some of the updated tests to fail until https://github.com/apache/tvm/pull/12724 lands.

Reply:
Forgot to mark this conversation as resolved earlier.